plugin/src/main/java/com/yelp/codegen/plugin/GenerateTask.kt

```diff
@@ -30,44 +33,51 @@ open class GenerateTask : DefaultTask() {
     @Input
     @Optional
     @Option(option = "platform", description = "Configures the platform that is used for generating the code.")
-    var platform: String? = null
+    var platformProvider: Provider<String> = project.objects.property(String::class.javaObjectType)
```

Comment:
Name should not be changed here as this is part of the public API (same for all of them).